### PR TITLE
Chunk insert all

### PR DESCRIPTION
**Pull request by jotaviobiondo**
Instead of querying the database to create the `Version` maps and then pass it to `insert_all` to create the versions, we now pass the query directly to the `insert_all` (new feature in Ecto 3.6), so we don't have problems with the Postgres parameters limit (when there's too many versions) and it's all happening in one query now.

### CODE REVIEW

File: config/test.exs [hunk 1]
```diff
@@ -6,18 +6,20 @@ config :paper_trail, repo: PaperTrail.Repo, originator: [name: :user, model: Use
 
 config :paper_trail, PaperTrail.Repo,
   adapter: Ecto.Adapters.Postgres,
-  username: System.get_env("POSTGRES_USER"),
-  password: System.get_env("POSTGRES_PASSWORD"),
+  username: System.get_env("POSTGRES_USER") || "postgres",
```
Comment (gabrielpra1) [inline config/test.exs:9]: Are we not using `System.get_env/2` to support older elixir versions?

File: lib/paper_trail/multi.ex [hunk 2]
```diff
@@ -162,8 +162,8 @@ defmodule PaperTrail.Multi do
 
       _ ->
         multi
+        |> Ecto.Multi.insert_all(version_key, Version, entries_query, returning: returning)
```
Comment (gabrielpra1) [inline lib/paper_trail/multi.ex:165]: just curious: is there a reason for inserting Versions before doing the update?

File: lib/paper_trail/serializer.ex [hunk 3]
```diff
@@ -178,25 +175,23 @@ defmodule PaperTrail.Serializer do
     {field, serialize(value, options, event)}
   end
 
+  defp dump_field!({field, [%Ecto.Changeset{} | _] = changesets}, _schema, _adapter, options, event) do
```
Comment (gabrielpra1) [inline lib/paper_trail/serializer.ex:178]: can you clarify this new `dump_field!` clause and the removal of `do_dump_field!`? Are they related?

**Review by gabrielpra1 — CHANGES_REQUESTED** [review]

File: lib/paper_trail/multi.ex (hunk 2 above)
Comment (jotaviobiondo) [reply lib/paper_trail/multi.ex:165]: Yes, this is a test for the case: https://github.com/jungsoft/paper_trail/pull/2/files#diff-37c9c2e2780268ed3baadb82751de315360e4ce88d02bb6d085e435c694fbd64R677


Basically it's because when the queryable given to the `update_all` use a field in the where clause that is modified by the `set` in the update. So if we have a query for users with `where is_active == false` and then `set: [is_activ: true]`, the result of the query will change after the update, so the `insert_all` of the versions will not function properly. 

Do you see a problem with this approach? Another approach would be first query for the ids of the changed records

File: lib/paper_trail/serializer.ex (hunk 3 above)
Comment (jotaviobiondo) [reply lib/paper_trail/serializer.ex:178]: Yes, they are related. The `do_dump_field!` was a quick workaround, but it introduced a bug (the tests caught):

```elixir
  defp do_dump_field!(schema, _field, {_, Ecto.Embedded, _}, _value, _adapter) do
    Ecto.embedded_dump(schema.__struct__(), :json)
  end
```

Notice that the `schema.__struct__()` is wrong. First because the `schema` we receive here is not the embedded schema, its the "main" schema. The second problem is that calling `__struct__()` will create a struct with nil values, thus, not serializing any changes. For example, in the tests we have a `Company` with a `EmailOptions` embedded schema. But the tests are failing because the serialized company version was wrong:

image

Basically this workaround was a fix for when we use `embeds_many` with `cast_embed` or `put_embed` and in the changes of the "main" changeset, there's a list of changesets for the embeds, but this case we need to handle "manually", otherwise this list of changesets is passed to the `Ecto.Type.adapter_dump(adapter, type, value)`, which raises an error since it cannot serialize a changesets.

Comment (gabrielpra1) [reply lib/paper_trail/serializer.ex:178]: Got it, thanks!
